docs/csharp/programming-guide/concepts/attributes/index.md

```diff
@@ -1,171 +1,124 @@
-  
- To explicitly identify an attribute target, use the following syntax:  
-  
-```csharp  
-[target : attribute-list]  
-```  
-  
- The list of possible `target` values is shown in the following table.  
-  
-|Target value|Applies to|  
-|------------------|----------------|  
-|`assembly`|Entire assembly|  
-|`module`|Current assembly module|  
-|`field`|Field in a class or a struct|  
-|`event`|Event|  
-|`method`|Method or `get` and `set` property accessors|  
-|`param`|Method parameters or `set` property accessor parameters|  
-|`property`|Property|  
-|`return`|Return value of a method, property indexer, or `get` property accessor|  
-|`type`|Struct, class, interface, enum, or delegate|  
-  
- The following example shows how to apply attributes to assemblies and modules. For more information, see [Common Attributes (C#)](../../../../csharp/programming-guide/concepts/attributes/common-attributes.md).  
-  
-```csharp  
-using System;  
-using System.Reflection;  
-[assembly: AssemblyTitleAttribute("Production assembly 4")]  
-[module: CLSCompliant(true)]  
-```  
-  
- The following example shows how to apply attributes to methods, method parameters, and method return values in C#.  
-  
-```csharp  
-// default: applies to method  
-[SomeAttr]  
-int Method1() { return 0; }  
-  
-// applies to method  
-[method: SomeAttr]  
-int Method2() { return 0; }  
-  
-// applies to return value  
-[return: SomeAttr]  
-int Method3() { return 0; }  
-```  
-  
+> By convention, all attribute names end with the word "Attribute" to distinguish them from other items in the .NET Framework. However, you do not need to specify the attribute suffix when using attributes in code. For example, `[DllImport]` is equivalent to `[DllImportAttribute]`, but `DllImportAttribute` is the attribute's actual name in the .NET Framework.
+
+### Attribute parameters
+
+Many attributes have parameters, which can be positional, unnamed, or named. Any positional parameters must be specified in a certain order and cannot be omitted. Named parameters are optional and can be specified in any order. Positional parameters are specified first. For example, these three attributes are equivalent:
+
+```csharp
+[DllImport("user32.dll")]
+[DllImport("user32.dll", SetLastError=false, ExactSpelling=false)]
+[DllImport("user32.dll", ExactSpelling=false, SetLastError=false)]
+```
+
+The first parameter, the DLL name, is positional and always comes first; the others are named. In this case, both named parameters default to false, so they can be omitted. Refer to the individual attribute's documentation for information on default parameter values.
```

Comment:
In terms of "Refer to...", the problem is that you have to know that the positional parameters correspond to an attribute constructor, and named/optional parameters to either properties or fields. That should probably be specified here.